extensions/gossip/api/gossipapi.go

```diff
@@ -42,3 +42,11 @@ type BlockPublisher interface {
 	// Publish traverses the block and invokes all applicable handlers
 	Publish(block *cb.Block)
 }
+
+// TxMetadata contain txn metadata
+type TxMetadata struct {
+	BlockNum  uint64
+	ChannelID string
+	ID        string
+	Num       uint64
```

Comment:
This structure could use a comment, as BlockNum vs Num will be confusing.

These together are referring to Height (https://github.com/hyperledger/fabric/blob/bbd0324b247e9e9d24ff445c7234225aaf552f4d/core/ledger/kvledger/txmgmt/version/version.go#L26)